src/GrammarParser.php

```diff
@@ -138,35 +160,34 @@ public function buildGrammar($grammarStr, $options = [])
         return $grammar;
     }
 
-    public function getParser()
+    public function getParser(): Parser
     {
         if ($this->parserSchouldBeRefreshed) {
-            $this->generateNewParser();
+            $this->parser = $this->generateNewParser();
```

Comment:
I decided to make the behaviour here explicit so reading the code makes it clear that `->parser` is explicitly set (refreshed) not implicitly in the method.